pyccel/codegen/printing/fcode.py

```diff
@@ -2215,7 +2216,13 @@ def _print_NumpyUfuncBase(self, expr):
             func_name = numpy_ufunc_to_fortran[type_name]
         except KeyError:
             errors.report(PYCCEL_RESTRICTION_TODO, severity='fatal')
-        code_args = ', '.join(self._print(i) for i in expr.args)
+        args = []
+        for arg in expr.args:
+            if arg.dtype is NativeInteger():
+                args.append(self._print(NumpyFloat(arg)))
+            else :
+                args.append(self._print(arg))
```

Comment:
This could be re-written with a list comprehension as:
```
args = [self_print(NumpyFloat(a) if a.dtype is NativeInteger() else a) for a in expr.args]
```
The line is a bit long so it might want breaking into 2, ie.:
```
args = [self_print(NumpyFloat(a) if a.dtype is NativeInteger() \
                            else a) for a in expr.args]
```
but list comprehension is much faster than a loop and append

Reply:
done